services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/background_task_periodic_heartbeat_check_setup.py

```diff
@@ -0,0 +1,74 @@
+import logging
+from collections.abc import Awaitable, Callable
+from typing import TypedDict
+
+from fastapi import FastAPI
+from servicelib.background_task import stop_periodic_task
+from servicelib.logging_utils import log_catch, log_context
+from servicelib.redis_utils import start_exclusive_periodic_task
+
+from ..core.settings import ApplicationSettings
+from .background_task_periodic_heartbeat_check import (
+    periodic_check_of_running_services_task,
+)
```

Comment:
To conclude my point from before.

I would have used only 1 module, where you put all your logic (task + setup).
You are creating a module whose task is `heartbeat_check` you can argue if `periodic` should be part of its name.

This way you create something that's called heartbeat_check from where you import `setup_heartbeat_check` which is very similar to other patterns we use elsewhere.

This way you also hide the implementation behind, which can change based on current requirements.